eventforwarder/forward.go

```diff
@@ -0,0 +1,100 @@
+package eventforwarder
+
+import (
+	"github.com/topfreegames/extensions/pg/interfaces"
+	"github.com/topfreegames/maestro/models"
+	"k8s.io/client-go/kubernetes"
+)
+
+func getEnabledForwarders(
+	schedulerForwarders map[string]map[string]*models.Forwarder,
+	configuredForwarders []*Info,
+) []EventForwarder {
+	enabledForwarders := make([]EventForwarder, 0)
+	for _, configuredFwdInfo := range configuredForwarders {
+		if schedulerFwds, ok := schedulerForwarders[configuredFwdInfo.Plugin]; ok {
+			if fwd, ok := schedulerFwds[configuredFwdInfo.Name]; ok {
+				if fwd.Enabled {
+					enabledForwarders = append(enabledForwarders, configuredFwdInfo.Forwarder)
+				}
+			}
+		}
+	}
+	return enabledForwarders
+}
+
+// ForwardRoomEvent forwards room event to app eventforwarders
+func ForwardRoomEvent(
+	forwarders []*Info,
+	db interfaces.DB,
+	kubernetesClient kubernetes.Interface,
+	room *models.Room,
+	status string,
+	metadata map[string]interface{},
+	schedulerCache *models.SchedulerCache,
+) error {
+	if len(forwarders) > 0 {
+		scheduler, err := schedulerCache.LoadScheduler(db, room.SchedulerName, true)
+		if err != nil {
+			return err
+		}
+		config, err := models.NewConfigYAML(scheduler.YAML)
+		if err != nil {
+			return err
+		}
+		infos, err := room.GetRoomInfos(db, kubernetesClient, schedulerCache, scheduler)
+		if err != nil {
+			return err
+		}
+		infos["metadata"] = metadata
+		if len(config.Forwarders) > 0 {
+			enabledForwarders := getEnabledForwarders(config.Forwarders, forwarders)
+			if len(enabledForwarders) > 0 {
+				return ForwardEventToForwarders(enabledForwarders, status, infos)
+			}
+		}
+	}
+	return nil
+}
+
+// ForwardPlayerEvent forwards player event to app eventforwarders
+func ForwardPlayerEvent(
+	forwarders []*Info,
+	db interfaces.DB,
+	kubernetesClient kubernetes.Interface,
+	room *models.Room,
+	event string,
+	metadata map[string]interface{},
+	schedulerCache *models.SchedulerCache,
+) error {
+	if len(forwarders) > 0 {
+		metadata["roomId"] = room.ID
+		scheduler, err := schedulerCache.LoadScheduler(db, room.SchedulerName, true)
+		if err != nil {
+			return err
+		}
+		metadata["game"] = scheduler.Game
+		config, err := models.NewConfigYAML(scheduler.YAML)
```

Comment:
It would be better to use schedulerCache's method LoadConfigYaml. This save us another Unmarshal.

Reply:
I'll change it too.